tests/unit/machine/EnsembleMachine_unittest.cc

```diff
@@ -77,13 +78,15 @@ TEST(combinate_composite_and_pipeline, train)
 	auto ground_truth =
 	    std::static_pointer_cast<MulticlassLabels>(mockData->get_labels_test());
 	auto pipeline = std::make_shared<PipelineBuilder>();
-	auto pred = pipeline ->over(std::make_shared<NormOne>())
-			 	        	->composite()
-				 	        	->over(std::make_shared<MulticlassLibLinear>())
-	            				->over(std::make_shared<MulticlassOCAS>())
-	                        	->then(std::make_shared<MeanRule>())
-	                	    		->train(train_feats, train_labels)
-	                		   		->apply_multiclass(test_feats);
+	auto machine = pipeline->over(std::make_shared<NormOne>())
+		 	        	->composite()
+			 	        	->over(std::make_shared<MulticlassLibLinear>())
+            				->over(std::make_shared<MulticlassOCAS>())
+                        	->then(std::make_shared<MeanRule>());
+	
+	machine->train(train_feats, train_labels);
+
+	auto pred = machine->apply_multiclass(test_feats);
```

Comment:
At some point Machines should return a pointer to themselves after calling train (like in sklearn) so that we can use a builder pattern in situations like this